### PR TITLE
Deprovision button on transfer history

### LINKED ISSUE #5: Use Managed Resources for Blob Transfers
## Description

When creating a data transfer where the destination is Azure storage, the destination container is currently hardcoded, which leads to issues when performing multiple data transfers.

Currently the result is overwritten, and the process indicator is no good since the .complete blob already exists.

With `managedResources=true` a new container would be created each time.

We should therefore omit the destination container and use `managedResources=true`

The container name (`src-container`) should not be sent in the request.

## Acceptance Criteria

- [ ] A new destination container is created for each data transfer.

### LINKED ISSUE #14: Deprovision Blob Storage Destination Container

### PR DESCRIPTION
**Pull request by marcgs**
closes agera-edc/DataDashboardFork#14 

Implementation of deprovision action in the transfer history table

Sample deployment: http://uitest485-company2-mvd.northeurope.azurecontainer.io

### CODE REVIEW

**Review by chrislomonico — APPROVED** [review]

File: src/modules/edc-demo/components/transfer-history/transfer-history-viewer.component.html [hunk 1]
```diff
@@ -50,15 +40,15 @@
...
         <ng-container matColumnDef="action">
-          <th mat-header-cell *matHeaderCellDef scope="col">Action</th>
-          <td mat-cell *matCellDef="let item">
-            <a *ngIf="item.dataDestination.properties.type == 'AzureStorage'" mat-button color="accent" [href]="storageExplorerLinkTemplate | replace: item.dataDestination.properties | safe: 'url'">Open Storage Explorer</a>
+          <th mat-header-cell *matHeaderCellDef scope="col">Actions</th>
+          <td mat-cell *matCellDef="let item" class="singleline-cell">
+            <a *ngIf="showStorageExplorerLink(item)" mat-stroked-button mat-button color="accent" [href]="storageExplorerLinkTemplate | replace: item.dataDestination.properties | safe: 'url'" class="action-button">
+              <mat-icon>loupe</mat-icon> Open Storage Explorer
+            </a>
+            <button *ngIf="showDeprovisionButton(item)" (click)="onDeprovision(item)" color="accent" mat-stroked-button  class="action-button">
```
Comment (paullatzelsperger) [inline src/modules/edc-demo/components/transfer-history/transfer-history-viewer.component.html:49]:
```suggestion
            <button *ngIf="showDeprovisionButton(item)" (click)="onDeprovision(item)" color="warn" mat-stroked-button  class="action-button">
```

Comment (paullatzelsperger) [reply src/modules/edc-demo/components/transfer-history/transfer-history-viewer.component.html:49]: maybe even add a confirmation dialog.

**Review by paullatzelsperger — CHANGES_REQUESTED** [review]
when the deprovisioning starts, the status changes to `DEPROVISIONING`, but when the deprovisioning is completed, the status does not get updated, so the user must manually reload the page.

**Comment by marcgs** [comment]
> when the deprovisioning starts, the status changes to `DEPROVISIONING`, but when the deprovisioning is completed, the status does not get updated, so the user must manually reload the page.

That is true, but the alternative of periodically reloading the table does not look very user friendly to me. Probably a refresh button would be better. I suggest to leave this as a possible future improvement.

**Comment by paullatzelsperger** [comment]
> > when the deprovisioning starts, the status changes to `DEPROVISIONING`, but when the deprovisioning is completed, the status does not get updated, so the user must manually reload the page.
> 
> That is true, but the alternative of periodically reloading the table does not look very user friendly to me. Probably a refresh button would be better. I suggest to leave this as a possible future improvement.

just quickly checking yielded that one needs a `Subject`, e.g. a `BehaviourSubject`, because the template binds to the instance of the `Observable` on page init, so if we reassign the instance to another one, it won't work anymore. There are examples of this in the code base (search for `fetch$`).

**Review by paullatzelsperger — APPROVED** [review]

**Pull request merged**